### ISSUE TITLE
[Quantization] What will happen if I delete some lines of calibration file?

### ISSUE THREAD

**Issue by WeixiangXu** [NVIDIA/TensorRT#3092]
I have a calibration file obtained with PTQ. It have several lines like this:

> TRT-8401-EntropyCalibration2
image: 3cd7f65a
batch_norm_0.tmp_2: 3e22ff2b
sigmoid_0.tmp_0: 3c010a14
p2o.Mul.1: 3e09c911
....
....

If I delete some lines of it, whether (a) it will fallback to FP32 to obtain higher accuracy or (b) it will randomly generate scales for the deleted lines ?

**Comment by zerollzeng** [comment]
You can quickly test this trtexec, I think the program will crash unless you specify the corresponding layer precision to FP32/FP16, but in this case you don't need to modify the calibration file(I strongly recommend not to do that)

**Comment by WeixiangXu** [comment]
Thanks for your kindly reply! I tested it with trtexec. After obtaining the calibration table through TensorRT-PTQ, I deleted some of the layers from it (image, batch_norm, etc.). Although trtexec can perform inference normally, the inference accuracy drops to 0. Based on this, I have a question: if the calibration table obtained through **pytorch-qat** is missing some of the layers required by TensorRT, it is likely to cause a decrease in accuracy. Therfore, how can I determine which layers' quantization tables are essential for TensorRT? @zerollzeng

**Comment by zerollzeng** [comment]
> if the calibration table obtained through **pytorch-qat** is missing some of the layers required by TensorRT, it is likely to cause a decrease in accuracy. Therfore, how can I determine which layers' quantization tables are essential for TensorRT?

If you are using QAT, then the scale should be in the Q/DQ layers so you don't need to calibration cache.
